lib/perl/Genome/Config/AnalysisProject/Command/Reprocess.pm

```diff
@@ -35,12 +42,23 @@ sub valid_statuses {
 sub execute {
     my $self = shift;
 
-    my $analysis_project = $self->analysis_project;
-    for my $bridge ($analysis_project->analysis_project_bridges) {
+    for my $bridge ( $self->_resolve_analysis_project_bridges ) {
         $bridge->reschedule;
     }
 
     return 1;
 }
 
+sub _resolve_analysis_project_bridges {
+    my $self = shift;
+
+    return $self->analysis_project->analysis_project_bridges if not $self->instrument_data;
+
+    return Genome::Config::AnalysisProject::InstrumentDataBridge->get(
+        analysis_project => $self->analysis_project,
+        'instrument_data.id' => [ map { $_->id}  $self->instrument_data ],
```

Comment:
:smile:, but it does an extra join!

Another option more parallel to the "all" case might be:
```perl
return $self->analysis_project->analysis_project_bridges(instrument_data_id => [map $_->id, $self->instrument_data]);
```